lib/SQL/Translator/Parser/DBIx/Class.pm

```diff
@@ -149,13 +149,16 @@ sub parse {
 
         $table->primary_key(@primary) if @primary;
 
-        my %unique_constraints = $source->unique_constraints;
-        foreach my $uniq (sort keys %unique_constraints) {
-            if (!$source->_compare_relationship_keys($unique_constraints{$uniq}, \@primary)) {
+        my $unique_constraints = $source->unique_constraints_info;
+        foreach my $uniq (sort keys %$unique_constraints) {
+            my $cons = $unique_constraints->{$uniq};
```

Comment:
Don't add extremely short-lived intermediate variables like that, it makes code harder to reason about, as in *why did he save it only to use it a couple times in this function, what am I missing*